### ISSUE TITLE
Potential implementation for space.periodic_general

### ISSUE THREAD

**Issue by evenmn** [google/jax-md#220]
Hi,

First, thanks for a great and intuitive MD package. 

I'm currently trying to implement the Vashishta potential in JAX-MD. My implementation seems to work for open boundaries, and potential energies agree with LAMMPS for all tested structures and parameters. However, when switching to triclinic boxes and periodic boundary conditions, my implementation doesn't seem to behave as expected. Where LAMMPS gives reasonable forces and energies, my JAX-MD implementation outputs energies up to 10^7 eV and forces up to 10^5 eV/Å. If I extend the box (with for instance 5Å in all directions), the range of the potential energy and forces seems reasonable again (but of course they are not correct). To me, it's clear that atoms overlap on the boundaries, even though they shouldn't given the cell.

Do I need to change anything in the implementation to account for the periodic boundaries? Or is there any obvious pitfalls to watch out for when implementing new potentials in JAX-MD? All advices that may push me in the right direction are highly appreciated.

**Comment by sirmarcel** [comment]
Is your cutoff larger than half the diameter of the simulation box? If yes, that's likely the cause of the issue, as `jax-md` will only loop over neighbours once, and does not take multiple replicas into account. (As far as I know, @sschoenholz will know for sure.) You can test by computing forces for a large supercell!

**Comment by evenmn** [comment]
Thank you for your advice, Marcel. This was indeed something that I didn't think about. One my systems has a simulation cell of
``` python
[[13.86135509  0.          0.        ]
 [-5.70404193 12.59861006  0.        ]
 [ 0.06201203 -0.48180903 13.02140076]]
```
while the cutoff is 5.5 (all in units of Å). The cutoff is therefore just smaller than half of the shortest box length. I tried to decrease the cutoff to 2Å, and I also tried to replicate the system in all directions but I still get a way too high energy (~1e4 eV for this particular system, LAMMPS gives around -1e2 eV).
